BUG: sparse.csgraph: Test failures with sparse 0.15.1 installed.

When I run the unit tests with `sparse` 0.15.1 installed (and not the latest 0.15.4), a bunch of `sparse` tests in `scipy/sparse/csgraph/tests/test_pydata_sparse.py` fail:

<details>
<summary>Test run transcript.</summary>

```
$ SCIPY_XSLOW=1 python3 dev.py test -t scipy/sparse/csgraph/tests/test_pydata_sparse.py -m full -j 10
💻  ninja -C /home/warren/repos/git/forks/scipy/build -j12
ninja: Entering directory `/home/warren/repos/git/forks/scipy/build'
[2/2] Generating scipy/generate-version with a custom command
Build OK
💻  meson install -C build --only-changed --tags runtime,python-runtime,tests,devel
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages
Running tests for scipy version:1.15.0.dev0+811.78a67fc, installed at:/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy
=============================== test session starts ===============================
platform linux -- Python 3.12.3, pytest-8.1.1, pluggy-1.5.0
rootdir: /home/warren/repos/git/forks/scipy
configfile: pytest.ini
plugins: timeout-2.3.1, xdist-3.5.0, hypothesis-6.100.1, cov-5.0.0, anyio-4.4.0
10 workers [58 items]     
............xx.xx.xxxxxxxx.xx.FxxxFFFxFFxxFxFxxxxxxFxFxF.F                  [100%]
==================================== FAILURES =====================================
_______________ test_nonzero_fill_value[COO-inf-isposinf-dijkstra] ________________
[gw3] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        comp_func  = <function isposinf at 0x705564777e70>
        fill_value = inf
        func       = <cyfunction dijkstra at 0x7055503669b0>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:539: in scipy.sparse.csgraph._shortest_path.dijkstra
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = False
        copy_if_sparse = False
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        csr_output = True
        dense_output = False
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of inf.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>,)
        i          = 0
____________ test_nonzero_fill_value[COO-inf-isposinf-floyd_warshall] _____________
[gw0] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        comp_func  = <function isposinf at 0x73eb5116caf0>
        fill_value = inf
        func       = <cyfunction floyd_warshall at 0x73eb3d3468e0>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:318: in scipy.sparse.csgraph._shortest_path.floyd_warshall
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = True
        copy_if_sparse = False
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        csr_output = False
        dense_output = True
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of inf.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>,)
        i          = 0
______________ test_nonzero_fill_value[COO-nan-isnan-shortest_path] _______________
[gw5] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        comp_func  = <ufunc 'isnan'>
        fill_value = nan
        func       = <cyfunction shortest_path at 0x73e703d6a810>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:164: in scipy.sparse.csgraph._shortest_path.shortest_path
    ???
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of nan.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>,)
        i          = 0
______________ test_nonzero_fill_value[COO-nan-isnan-floyd_warshall] ______________
[gw2] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        comp_func  = <ufunc 'isnan'>
        fill_value = nan
        func       = <cyfunction floyd_warshall at 0x75d8eb36a8e0>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:318: in scipy.sparse.csgraph._shortest_path.floyd_warshall
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = True
        copy_if_sparse = False
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        csr_output = False
        dense_output = True
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of nan.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>,)
        i          = 0
_____________ test_nonzero_fill_value[COO-inf-isposinf-bellman_ford] ______________
[gw3] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        comp_func  = <function isposinf at 0x705564777e70>
        fill_value = inf
        func       = <cyfunction bellman_ford at 0x705550366a80>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:1025: in scipy.sparse.csgraph._shortest_path.bellman_ford
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = False
        copy_if_sparse = False
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        csr_output = True
        dense_output = False
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of inf.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>,)
        i          = 0
_________________ test_nonzero_fill_value[COO-nan-isnan-johnson] __________________
[gw1] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        comp_func  = <ufunc 'isnan'>
        fill_value = nan
        func       = <cyfunction johnson at 0x72c73c142b50>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:1268: in scipy.sparse.csgraph._shortest_path.johnson
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = False
        copy_if_sparse = False
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        csr_output = True
        dense_output = False
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of nan.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>,)
        i          = 0
_________ test_nonzero_fill_value[COO-inf-isposinf-minimum_spanning_tree] _________
[gw4] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        comp_func  = <function isposinf at 0x74a1282046f0>
        fill_value = inf
        func       = <cyfunction minimum_spanning_tree at 0x74a117f4f370>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_min_spanning_tree.pyx:101: in scipy.sparse.csgraph._min_spanning_tree.minimum_spanning_tree
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = False
        copy_if_sparse = True
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        csr_output = True
        dense_output = False
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of inf.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>,)
        i          = 0
_________________ test_nonzero_fill_value[COO-nan-isnan-dijkstra] _________________
[gw0] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        comp_func  = <ufunc 'isnan'>
        fill_value = nan
        func       = <cyfunction dijkstra at 0x73eb3d3469b0>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:539: in scipy.sparse.csgraph._shortest_path.dijkstra
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = False
        copy_if_sparse = False
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        csr_output = True
        dense_output = False
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of nan.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>,)
        i          = 0
__________ test_nonzero_fill_value[COO-nan-isnan-minimum_spanning_tree] ___________
[gw6] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        comp_func  = <ufunc 'isnan'>
        fill_value = nan
        func       = <cyfunction minimum_spanning_tree at 0x7575cfd67370>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_min_spanning_tree.pyx:101: in scipy.sparse.csgraph._min_spanning_tree.minimum_spanning_tree
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = False
        copy_if_sparse = True
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        csr_output = True
        dense_output = False
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of nan.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>,)
        i          = 0
_______________ test_nonzero_fill_value[COO-nan-isnan-bellman_ford] _______________
[gw4] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        comp_func  = <ufunc 'isnan'>
        fill_value = nan
        func       = <cyfunction bellman_ford at 0x74a117f4ea80>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:1025: in scipy.sparse.csgraph._shortest_path.bellman_ford
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = False
        copy_if_sparse = False
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        csr_output = True
        dense_output = False
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of nan.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=nan>,)
        i          = 0
_____________ test_nonzero_fill_value[COO-inf-isposinf-shortest_path] _____________
[gw8] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        comp_func  = <function isposinf at 0x7dc061b88cb0>
        fill_value = inf
        func       = <cyfunction shortest_path at 0x7dc0568ce810>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:164: in scipy.sparse.csgraph._shortest_path.shortest_path
    ???
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of inf.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>,)
        i          = 0
________________ test_nonzero_fill_value[COO-inf-isposinf-johnson] ________________
[gw7] linux -- Python 3.12.3 /home/warren/py3.12.3/bin/python3
scipy/sparse/_sputils.py:405: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse(accept_fv=accept_fv)
E   TypeError: COO.to_scipy_sparse() got an unexpected keyword argument 'accept_fv'
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None

During handling of the above exception, another exception occurred:
scipy/sparse/csgraph/tests/test_pydata_sparse.py:173: in test_nonzero_fill_value
    actual = func(A_sparse)
        A_dense    = array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]])
        A_sparse   = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        comp_func  = <function isposinf at 0x72b6a4470630>
        fill_value = inf
        func       = <cyfunction johnson at 0x72b697146b50>
        graphs     = (array([[0, 1, 1, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 1],
       [0, 0, 0, 0, 0]]), <COO: shape=(5, 5), dtype=int64, nnz=4, fill_value=0>)
        sparse_cls = <class 'sparse._coo.core.COO'>
_shortest_path.pyx:1268: in scipy.sparse.csgraph._shortest_path.johnson
    ???
scipy/sparse/csgraph/_validation.py:26: in validate_graph
    csgraph = convert_pydata_sparse_to_scipy(csgraph, accept_fv=accept_fv)
        accept_fv  = [0, inf, nan]
        copy_if_dense = False
        copy_if_sparse = False
        csgraph    = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        csr_output = True
        dense_output = False
        directed   = True
        dtype      = <class 'numpy.float64'>
        infinity_null = True
        nan_null   = True
        null_value_in = 0
        null_value_out = inf
scipy/sparse/_sputils.py:407: in convert_pydata_sparse_to_scipy
    arg = arg.to_scipy_sparse()
        accept_fv  = [0, inf, nan]
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        target_format = None
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_coo/core.py:1180: in to_scipy_sparse
    check_zero_fill_value(self)
        self       = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
/home/warren/py3.12.3/lib/python3.12/site-packages/sparse/_utils.py:558: in check_zero_fill_value
    raise ValueError(
E   ValueError: This operation requires zero fill values, but argument 0 had a fill value of inf.
        arg        = <COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>
        args       = (<COO: shape=(5, 5), dtype=float64, nnz=4, fill_value=inf>,)
        i          = 0
============================= short test summary info =============================
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-inf-isposinf-dijkstra] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-inf-isposinf-floyd_warshall] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-nan-isnan-shortest_path] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-nan-isnan-floyd_warshall] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-inf-isposinf-bellman_ford] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-nan-isnan-johnson] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-inf-isposinf-minimum_spanning_tree] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-nan-isnan-dijkstra] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-nan-isnan-minimum_spanning_tree] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-nan-isnan-bellman_ford] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-inf-isposinf-shortest_path] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
FAILED scipy/sparse/csgraph/tests/test_pydata_sparse.py::test_nonzero_fill_value[COO-inf-isposinf-johnson] - ValueError: This operation requires zero fill values, but argument 0 had a fil...
==================== 12 failed, 17 passed, 29 xfailed in 3.49s ====================

```

</details>

It looks like the problem is here:

https://github.com/scipy/scipy/blob/78a67fc32d3502d3d6982e731a5f0a97c18ab37b/scipy/sparse/_sputils.py#L402-L407

The expected exception is raised in line 405, but then another exception is raised in line 407 because an underlying function does not handle a nonzero fill value.

--- a/scipy/sparse/csgraph/tests/test_pydata_sparse.py
+++ b/scipy/sparse/csgraph/tests/test_pydata_sparse.py
@@ -3,6 +3,7 @@ import pytest
 import numpy as np
 import scipy.sparse as sp
 import scipy.sparse.csgraph as spgraph
+from scipy._lib import _pep440
 
 from numpy.testing import assert_equal
 
@@ -20,6 +21,15 @@ msg = "pydata/sparse (0.15.1) does not implement necessary operations"
 
 sparse_params = (pytest.param("COO"),
                  pytest.param("DOK", marks=[pytest.mark.xfail(reason=msg)]))
+
+
+def check_sparse_version(min_ver):
+    if sparse is None:
+        return pytest.mark.skip(reason="sparse is not installed")
+    return pytest.mark.skipif(
+        _pep440.parse(sparse.__version__) < _pep440.Version(min_ver),
+        reason=f"sparse version >= {min_ver} required"
+    )
 
 
 @pytest.fixture(params=sparse_params)
@@ -149,6 +159,7 @@ def test_min_weight_full_bipartite_matching(graphs):
     assert_equal(actual, desired)
 
 
+@check_sparse_version("0.15.4")
 @pytest.mark.parametrize(
     "func",
     [

